cpp/src/search/search.cu

```diff
@@ -104,39 +125,18 @@ std::unique_ptr<column> search_ordered(table_view const& t,
 
   // This utility will ensure all corresponding dictionary columns have matching keys.
   // It will return any new dictionary columns created as well as updated table_views.
-  auto const matched = dictionary::detail::match_dictionaries({t, values}, stream);
-
-  // Prepare to flatten the structs column
-  auto const has_null_elements   = has_nested_nulls(t) or has_nested_nulls(values);
-  auto const flatten_nullability = has_null_elements
-                                     ? structs::detail::column_nullability::FORCE
-                                     : structs::detail::column_nullability::MATCH_INCOMING;
-
-  // 0-table_view, 1-column_order, 2-null_precedence, 3-validity_columns
-  auto const t_flattened = structs::detail::flatten_nested_columns(
-    matched.second.front(), column_order, null_precedence, flatten_nullability);
-  auto const values_flattened =
-    structs::detail::flatten_nested_columns(matched.second.back(), {}, {}, flatten_nullability);
-
-  auto const t_d      = table_device_view::create(t_flattened, stream);
-  auto const values_d = table_device_view::create(values_flattened, stream);
-  auto const& lhs     = find_first ? *t_d : *values_d;
-  auto const& rhs     = find_first ? *values_d : *t_d;
-
-  auto const& column_order_flattened    = t_flattened.orders();
-  auto const& null_precedence_flattened = t_flattened.null_orders();
-  auto const column_order_dv = detail::make_device_uvector_async(column_order_flattened, stream);
-  auto const null_precedence_dv =
-    detail::make_device_uvector_async(null_precedence_flattened, stream);
-
-  auto const count_it = thrust::make_counting_iterator<size_type>(0);
-  auto const comp     = row_lexicographic_comparator(nullate::DYNAMIC{has_null_elements},
-                                                 lhs,
-                                                 rhs,
-                                                 column_order_dv.data(),
-                                                 null_precedence_dv.data());
-  launch_search(
-    count_it, count_it, t.num_rows(), values.num_rows(), result_out, comp, find_first, stream);
+  auto const matched        = dictionary::detail::match_dictionaries({t, values}, stream);
+  auto const matched_t      = matched.second.front();
+  auto const matched_values = matched.second.back();
+
+  auto const& lhs       = find_first ? matched_t : matched_values;
+  auto const& rhs       = find_first ? matched_values : matched_t;
+  auto const comparator = cudf::experimental::row::lexicographic::two_table_comparator(
+    lhs, rhs, column_order, null_precedence, stream);
```

Comment:
Since we don't care about table order:
```suggestion
  auto const comparator = cudf::experimental::row::lexicographic::two_table_comparator(
    matched_t, matched_values, column_order, null_precedence, stream);
```

Reply:
I just merged this PR and search tests failed with this fix, due to this => https://github.com/rapidsai/cudf/pull/10730/files#diff-0743b70282a92785dee8e55adc4099f501b4ed6c20c21e39fe187b2f4f0664a0R470.